src/js/control-bar/text-track-controls/chapters-button.js

```diff
@@ -37,108 +34,103 @@ class ChaptersButton extends TextTrackButton {
     return `vjs-chapters-button ${super.buildCSSClass()}`;
   }
 
-  /**
-   * Create a menu item for each text track
-   *
-   * @return {Array} Array of menu items
-   * @method createItems
-   */
-  createItems() {
-    const items = [];
-    const tracks = this.player_.textTracks();
+  update(event) {
+    if (this.track_ === undefined || (event && (event.type === 'addtrack' || event.type === 'removetrack'))) {
+      this.setTrack(this.findChaptersTrack());
+    }
+    super.update();
+  }
 
-    if (!tracks) {
-      return items;
+  setTrack(track) {
+    if (this.track_ === track) {
+      return;
     }
 
-    for (let i = 0; i < tracks.length; i++) {
-      const track = tracks[i];
+    if (!this.updateHandler_) {
+      this.updateHandler_ = this.update.bind(this);
+    }
 
-      if (track.kind === this.kind_) {
-        items.push(new TextTrackMenuItem(this.player_, {track}));
+    if (this.track_) {
```

Comment:
can we add a comment here that says that for this `if`, `this._track_` is still the old value?